addons/api/addon/models/host.js

```diff
@@ -5,6 +5,7 @@
 
 import GeneratedHostModel from '../generated/models/host';
 import { attr } from '@ember-data/model';
+import { pluginTypes } from './host-catalog';
```

Comment:
None blocking comment: is this what you are looking to refactor in the future with the `TYPE` thread we discuss yesterday on slack? [link here](https://hashicorp.slack.com/archives/C012GTH1C4E/p1694620307660139)

Reply:
Yes. Just with it being a larger refactor, I didn't want to include it in this PR as it would grow to 20+ files changed.

Reply:
Definetly and totally onboard with your opinion, I was just wondering after our chat 😝 . Good work!